spec/std/http/server/server_spec.cr

```diff
@@ -151,6 +151,15 @@ module HTTP
     end
   end
 
+  describe HTTP::Server do
+    it "re-sets special port zero after listening" do
+      server = Server.new("0.0.0.0", 0) { |ctx| }
+      spawn { server.listen }
+      sleep 0.0001
```

Comment:
I like it. My only worry is that now we'll have `start` and `listen` and users might not read the docs (which will exist some day) and do `start` and the program will exit and complain, only to find out there's a `listen` method. Maybe a different name like `bind` or `setup` or similar?